### PR TITLE
feat:[IE-400]: Update external secrets for ng-manager

### PR DESCRIPTION
**Pull request by Vbhatia20**
## Describe your changes

**Build**:
https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/home/orgs/default/projects/DevSpaces/pipelines/Build_Services/executions/i9AEEzWiQ92X_MePl65-Vw/pipeline?connectorRef=harnessplinfra&repoName=harness-pl-infra&branch=master&storeType=REMOTE

**Deployment**:
Screenshot 2023-09-26 at 1 56 26 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53399)

### CODE REVIEW

File: 120-ng-manager/chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -64,3 +64,21 @@ Create the name of the service account to use
...
+USAGE:
+{{- "ng-manager.generateSecrets" (dict "ctx" $)}}
+*/}}
+{{- define "ng-manager.generateSecrets" }}
+    {{- $ := .ctx }}
+    {{- $hasAtleastOneSecret := false }}
+    {{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+    {{- if eq (include "harnesscommon.secrets.isDefault" (dict "ctx" $ "variableName" "LOG_STREAMING_SERVICE_TOKEN" "extKubernetesSecretCtxs" (list $.Values.secrets.kubernetesSecrets) "esoSecretCtxs" (list (dict $localESOSecretCtxIdentifier $.Values.secrets.secretManagement.externalSecretsOperator)))) "true" }}
+    {{- $hasAtleastOneSecret = true }}
+LOG_STREAMING_SERVICE_TOKEN: c76e567a-b341-404d-a8dd-d9738714eb82
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/_helpers.tpl:79]: I don't see "LOG_STREAMING_SERVICE_TOKEN" being used by ng-manager previously.

File: 120-ng-manager/chart/templates/config.yaml
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/config.yaml:1]: GITOPS_SERVICE_SECRET should be moved to secret.

File: 120-ng-manager/chart/templates/_helpers.tpl (hunk 1 above)
Comment (Vbhatia20) [reply 120-ng-manager/chart/templates/_helpers.tpl:79]: https://github.com/harness/harness-core/blob/be8a8e0c2971be0e53f86b85f300f0a48fc9635f/120-ng-manager/chart/templates/config.yaml#L41

Comment (Vbhatia20) [reply 120-ng-manager/chart/templates/_helpers.tpl:79]: it is used, this is from develop

**Comment by Vbhatia20** [comment]
trigger uts

File: 120-ng-manager/chart/templates/deployment.yaml [hunk 2]
```diff
@@ -80,20 +80,19 @@ spec:
...
-            {{- include "harnesscommon.dbconnection.timescaleEnv" (dict "passwordVariableName" "TIMESCALE_PASSWORD" "userVariableName" "TIMESCALEDB_USERNAME" "context" $) | indent 12}}
-            {{- include "harnesscommon.dbconnection.timescaleSslEnv" (dict "sslModeValue" "require" "certVariableName" "TIMESCALEDB_SSL_ROOT_CERT" "context" $ ) | nindent 12 }}
-            
+            {{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+            {{- include "harnesscommon.secrets.manageEnv" (dict "ctx" . "variableName" "LOG_STREAMING_SERVICE_TOKEN" "defaultKubernetesSecretName" "ng-manager-secret" "defaultKubernetesSecretKey" "LOG_STREAMING_SERVICE_TOKEN" "extKubernetesSecretCtxs" (list .Values.secrets.kubernetesSecrets) "esoSecretCtxs" (list (dict "secretCtxIdentifier" $localESOSecretCtxIdentifier "secretCtx" .Values.secrets.secretManagement.externalSecretsOperator))) | indent 12 }}
+            {{- include "harnesscommon.dbconnectionv2.redisEnv" (dict "ctx" . "localRedisCtx" .Values.redis "globalRedisCtx" .Values.global.database.redis) | indent 12 }}
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/deployment.yaml:85]: REDIS Env variables here are different than what we had implemented in template functions: https://github.com/harness/helm-common/blob/main/src/common/templates/_databaseconnectionsv2.tpl#L105. We will have to update the template functions to take the env variable names as input.

File: 120-ng-manager/chart/templates/deployment.yaml [hunk 3]
```diff
@@ -80,20 +80,19 @@ spec:
...
-            {{- include "harnesscommon.dbconnection.mongoEnv" . | indent 12 }}
-            {{- include "harnesscommon.dbconnection.timescaleEnv" (dict "passwordVariableName" "TIMESCALE_PASSWORD" "userVariableName" "TIMESCALEDB_USERNAME" "context" $) | indent 12}}
-            {{- include "harnesscommon.dbconnection.timescaleSslEnv" (dict "sslModeValue" "require" "certVariableName" "TIMESCALEDB_SSL_ROOT_CERT" "context" $ ) | nindent 12 }}
-            
+            {{- $localESOSecretCtxIdentifier := (include "harnesscommon.secrets.localESOSecretCtxIdentifier" (dict "ctx" $ )) }}
+            {{- include "harnesscommon.secrets.manageEnv" (dict "ctx" . "variableName" "LOG_STREAMING_SERVICE_TOKEN" "defaultKubernetesSecretName" "ng-manager-secret" "defaultKubernetesSecretKey" "LOG_STREAMING_SERVICE_TOKEN" "extKubernetesSecretCtxs" (list .Values.secrets.kubernetesSecrets) "esoSecretCtxs" (list (dict "secretCtxIdentifier" $localESOSecretCtxIdentifier "secretCtx" .Values.secrets.secretManagement.externalSecretsOperator))) | indent 12 }}
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/deployment.yaml:84]: We should do similarly for other service secrets and SMTP secrets.

File: 120-ng-manager/chart/values.yaml [hunk 4]
```diff
@@ -310,3 +452,88 @@ virtualService:
...
+        SMTP_USERNAME: ""
+    - secretName: ""
+      keys:
+        SMTP_PASSWORD: ""
+    - secretName: ""
+      keys:
+        SMTP_USE_SSL: ""
+    - secretName: ""
+      keys:
+        SMTP_PORT: ""
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/values.yaml:481]: Can we combine all secret variables under keys of one KubernetesSecrets object?

File: 120-ng-manager/chart/values.yaml [hunk 5]
```diff
@@ -310,3 +452,88 @@ virtualService:
...
+          kind: ""
+        remoteKeys:
+          GITOPS_SERVICE_SECRET:
+            name: ""
+            property: ""
+      - secretStore:
+          name: ""
+          kind: ""
+        remoteKeys:
+          SMTP_HOST:
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/values.yaml:509]: Can we combine all secret variables under remoteKeys of one externalSecretsOperator object?

**Review by saitejavn-harness — COMMENTED** [review]
Update helm-common to 1.1.1

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Review by harness-jc — APPROVED** [review]

**Comment by harness-jc** [comment]
trigger its

**Comment by harness-jc** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger uts

File: 120-ng-manager/chart/templates/deployment.yaml [hunk 6]
```diff
@@ -80,20 +80,27 @@ spec:
...
+            {{- include "harnesscommon.secrets.manageEnv" (dict "ctx" . "variableName" "GITOPS_SERVICE_SECRET" "defaultKubernetesSecretName" "ng-manager-secret" "defaultKubernetesSecretKey" "GITOPS_SERVICE_SECRET" "extKubernetesSecretCtxs" (list .Values.secrets.kubernetesSecrets) "esoSecretCtxs" (list (dict "secretCtxIdentifier" $localESOSecretCtxIdentifier "secretCtx" .Values.secrets.secretManagement.externalSecretsOperator))) | indent 12 }}
+            {{- include "harnesscommon.secrets.manageEnv" (dict "ctx" . "variableName" "SMTP_HOST" "defaultKubernetesSecretName" "ng-manager-secret" "defaultKubernetesSecretKey" "SMTP_HOST" "extKubernetesSecretCtxs" (list .Values.secrets.kubernetesSecrets) "esoSecretCtxs" (list (dict "secretCtxIdentifier" $localESOSecretCtxIdentifier "secretCtx" .Values.secrets.secretManagement.externalSecretsOperator))) | indent 12 }}
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/deployment.yaml:87]: Are SMTP secrets always created?

File: 120-ng-manager/chart/templates/config.yaml [hunk 7]
```diff
@@ -72,7 +70,6 @@ data:
   {{- end}}
```
Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/config.yaml:70]: Could you handle "SMP_LICENSE" as a secret? Reference: https://github.com/harness/harness-core/pull/52866/files#diff-851bd6b338c59bf8dccfc7662c274ad0dbdfec103d43a6f368a9af258acdddceR83

Comment (saitejavn-harness) [inline 120-ng-manager/chart/templates/config.yaml:70]: SMP_LICENSE can now be removed from configmap as it is being handled in deployment.yaml

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger uts

**Review by saitejavn-harness — APPROVED** [review]

**Comment by Vbhatia20** [comment]
trigger uts

**Comment by Vbhatia20** [comment]
trigger gitleaks

**Comment by Vbhatia20** [comment]
trigger uts

**Pull request closed without merging**